File: src/main/scala/uk/co/halfninja/randomnames/NameGenerator.scala
package uk.co.halfninja.randomnames

trait NameGenerator {
	def generate(gender: Gender): Name
	def generate(gender: Gender, seed: Long): Name
	def generate(gender: Gender, mother: Name, father: Name): Name
	def generate(gender: Gender, mother: Name, father: Name, seed: Long): Name
}

File: src/main/kotlin/uk/co/halfninja/randomnames/NameGenerator.kt
package uk.co.halfninja.randomnames

interface NameGenerator {
    fun generate(gender: Gender): Name
    fun generate(gender: Gender, seed: Long): Name
    fun generate(gender: Gender, mother: Name, father: Name): Name
    fun generate(gender: Gender, mother: Name, father: Name, seed: Long): Name
}
